### PR TITLE
Document support for `shouldResubscribe` function

### PR DESCRIPTION
**Pull request by oliverwoodings**
Docs update for this feature PR: https://github.com/apollographql/react-apollo/pull/1896/files

### CODE REVIEW

File: docs/source/advanced/subscriptions.md [hunk 1]
```diff
@@ -157,10 +157,12 @@ The Subscription component accepts the following props. Only `subscription` and
   <dd>A function returning the UI you want to render based on your subscription result. **Required**</dd>
   <dt>`variables`: { [key: string]: any }</dt>
   <dd>An object containing all of the variables your subscription needs to execute</dd>
-  <dt>`shouldResubscribe`: boolean</dt>
-  <dd>Determines if your subscription should be unsubscribed and subscribed again</dd>
+  <dt>`shouldResubscribe`: boolean or (currentProps: Object, nextProps: Object) => boolean</dt>
```
Comment (oliverwoodings) [inline docs/source/advanced/subscriptions.md]: @jbaxleyiii I wasn't entirely sure how you guys documented variable-type props. Is `or` good enough here?

Comment (Poincare) [reply docs/source/advanced/subscriptions.md]: I believe "|" would be more appropriate here, as you can see on [this page](https://www.apollographql.com/docs/react/api/apollo-client.html).

**Comment by oliverwoodings** [comment]
@Poincare done!

**Comment by Poincare** [comment]
The Apollo Client tests seem to be failing on CircleCI, which 99.99% likely is unrelated but I'll confirm before merging.

**Comment by oliverwoodings** [comment]
Must be that markdown cryptominer i snuck in there

**Pull request merged**